### PR TITLE
Kennric/app2 formsender

### PR DESCRIPTION
**Pull request by Kennric**
@ramereth @armiller adding formsender to app2 - I still need to test the gunicorn command in situ.

### CODE REVIEW

**Comment by ramereth** [comment]
@Kennric I'm assuming this probably should wait until #6 gets merged as it adds the basic pieces for app2?

**Comment by Kennric** [comment]
@ramereth I think this one can go first, as the app is simpler and it should be easier to test and debug - I'll rebase work in #6 when this is merged

**Comment by ramereth** [comment]
@Kennric please rebase from master and push it up and I'll review this properly now that #8 has been merged.

**Comment by Kennric** [comment]
@ramereth rebased

File: recipes/app2.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include_recipe 'osl-app::default'
+
+node.normal['users'] = %w(formsender-production, formsender-staging)
```
Comment (ramereth) [inline recipes/app2.rb]: Please remove the comma as its not needed.

File: recipes/app2.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  install do
+    wanted_by 'multi-user.target'
+  end
+  service do
+    type 'forking'
+    user 'formsender-staging'
+    environment 'PATH => /home/formsender-staging/venv/bin'
+    working_directory '/home/formsender-staging/formsender'
+    pid_file '/home/formsender-staging/tmp/pids/gunicorn.pid'
+    exec_start '/home/formsender-staging/venv/bin/gunicorn -b 0.0.0.0:8086 formsender.wsgi:application'
```
Comment (ramereth) [inline recipes/app2.rb]: This line is too long, please wrap it.

File: recipes/app2.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  install do
+    wanted_by 'multi-user.target'
+  end
+  service do
+    type 'forking'
+    user 'formsender-production'
+    environment 'PATH => /home/formsender-production/venv/bin'
+    working_directory '/home/formsender-production/formsender'
+    pid_file '/home/formsender-production/tmp/pids/gunicorn.pid'
+    exec_start '/home/formsender-production/venv/bin/gunicorn -b 0.0.0.0:8085 formsender.wsgi:application'
```
Comment (ramereth) [inline recipes/app2.rb]: This line is too long, please wrap it.

File: spec/app2_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                   '/usr/bin/systemctl stop formsender-production-gunicorn',
+                   '/usr/bin/systemctl start formsender-production-gunicorn',
+                   '/usr/bin/systemctl status formsender-production-gunicorn',
+                   '/usr/bin/systemctl reload formsender-production-gunicorn',
+                   '/usr/bin/systemctl restart formsender-production-gunicorn'],
+        nopasswd: true
+      )
+    end
+
+    %w(formsender-staging-gunicorn formsender-production-gunicorn ).each do |s|
```
Comment (ramereth) [inline spec/app2_spec.rb]: Please remove the space before ``)``

File: spec/app2_spec.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
+require 'chefspec'
+require 'chefspec/berkshelf'
+
+describe 'osl-app::app2' do
+  context 'on centos 7.2' do
+    let(:chef_run) do
+      ChefSpec::SoloRunner.new(platform: 'centos',
+                               version: '7.2.1511').converge(described_recipe)
+    end
```
Comment (ramereth) [inline spec/app2_spec.rb]: Take a look at what I did in ``app1_spec.rb`` and add it here so the test works.

**Comment by Kennric** [comment]
@ramereth  Tests now pass.

File: .kitchen.yml [hunk 6]
```diff
@@ -19,3 +19,4 @@ suites:
       - recipe[sudo]
       - recipe[user::data_bag]
       - recipe[osl-app::app1]
+      - recipe[osl-app::app2]
```
Comment (ramereth) [inline .kitchen.yml]: This shouldn't be int he app1 suite, but in its own app2 suite.

Comment (Kennric) [inline .kitchen.yml]: Oh, I missed that there were multiple suites. I see that the user attribute for the app1 suite is 'iam' - does this username matter? 'iam' is not a username we're actually using on app1 or for iam (we have iam-staging and iam-production)

Comment (ramereth) [inline .kitchen.yml]: Just create some additional databag users for what you need for testing.

**Comment by Kennric** [comment]
Tests pass. @ramereth

File: .kitchen.yml (hunk 6 above)
Comment (ramereth) [inline .kitchen.yml]: You still need to make the actual databags for each user in the repo for testing as TK fails when I try to run it.

**Comment by Kennric** [comment]
@ramereth data bags added, converges cleanly and tests pass.

**Comment by ramereth** [comment]
@Kennric please make sure you do ``!bump minor ~`` for merging this.

**Comment by Kennric** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v1.5.0.  Have a nice day!

**Pull request merged**